src/components/wizards/create-vm/CreateVMWizard.js

```diff
@@ -0,0 +1,366 @@
+        kind: 'VirtualMachine',
+        metadata : {
+          name: basicSettings.name.value,
+          labels: {
+            'kubevirt.io/os': basicSettings.operatingSystem.value
+          }
+        },
+        spec: {
+          template: {}
+        }
+      }
+      if(basicSettings.namespace){
+        vm.metadata.namespace = basicSettings.namespace.value;
+      }
+      if(basicSettings.description){
+        vm.metadata.labels.description = basicSettings.description;
+      }
+      if(basicSettings.flavor.value === 'Custom'){
+        vm.spec.template.spec = {
+          domain: {
+            cpu: {
+              cores: basicSettings.cpu.value
+            },
+            resources: {
+              requests: {
+                memory: `${basicSettings.memory.value}Gi`
+              }
+            }
+          }
+        }
+      } else {
+        vm.spec.template.spec = {
+          metadata: {
+            labels: {
+              'kubevirt.io/flavor': basicSettings.flavor.value
+            }
+          }
+        }
+      }
+      if(_.get(basicSettings.imageSourceType, 'value') === 'Registry'){
+        let domain = _.get(vm.spec.template.spec, 'domain', {});
+        let devices = _.get(domain,'devices',{});
+        let disks = _.get(devices,'disks',[]);
+        disks.push({
+          name: 'registryvolume',
+          volumeName: 'registryvolume',
+          disk: {
+            bus: 'virtio'
+          }
+        });
+        devices.disks = disks;
+        domain.devices = devices;
+        vm.spec.template.spec.domain = domain;
+
+        let volumes = _.get(vm.spec.template.spec, 'volumes', []);
+        volumes.push({
+          name: 'registryvolume',
+          registryDisk: {
+            image: basicSettings.registryImage.value
+          }
+        });
+        vm.spec.template.spec.volumes = volumes;
+      }
+      if(_.get(basicSettings.cloudInit, 'value', false)) {
+        let domain = _.get(vm.spec.template.spec, 'domain', {});
+        let devices = _.get(domain,'devices',[]);
+        devices.push({
+          name: 'cloudinitdisk',
+          volumeName: 'cloudinitdisk',
+          disk: {
+            bus: 'virtio'
+          }
+        });
+        domain.devices = devices;
+        vm.spec.template.spec.domain = domain;
+
+        let volumes = _.get(vm.spec.template.spec, 'volumes', []);
+        volumes.push({
+          name: 'cloudinitvolume',
+          cloudInitNoCloud: {
+            userData: {
+              'ssh-authorized-keys': [
+                `${basicSettings.authKeys.value} ${basicSettings.hostName.value}`
+              ]
+            }
+          }
+        });
+        vm.spec.template.spec.volumes = volumes;
+      }
+      if(basicSettings.startVM){
+        vm.spec.running=basicSettings.startVM.value;
+      }
+      return vm;
+    }
+  
+    closeWizard = () => {
+      this.setState(defaultState);
+      this.props.closeWizard();
```

Comment:
If `closeWizard` is meant to be a callback prop, please call it `onClose` instead.

Reply:
Also, patternfly-react `Wizard` already provides `onExited` callback prop that executes when the modal finishes its fade-out transition, I'd suggest using that instead.